065-topic-replication-factor-change.md

```diff
@@ -0,0 +1,298 @@
+| STRIMZI_CRUISE_CONTROL_AUTH_ENABLED   | Boolean | false   | Whether Basic authentication is enabled                       |
+
+With the Cluster Operator deployment, the certificate chain in PEM format and the admin user credentials for the Cruise Control REST API will be automatically mounted in the Topic Operator container.
+Instead, with the standalone deployment, the user will have to mount: certificate chain in `/etc/tls-sidecar/cluster-ca-certs/ca.crt`, credentials in `/etc/eto-cc-api/cruise-control.apiToAdminUsername` and `/etc/eto-cc-api/cruise-control.apiToAdminPassword`.
+
+### KafkaTopic CRD changes
+
+During the whole replicas change process the KafkaTopic will remain in a ready state.
+A replicas change will be executed asynchronously, taking more than one reconciliation to complete.
+
+A new optional KafkaTopicStatus subsection called `replicasChange` will be used to update the user, and keep track of the replicas change state.
+The `replicasChange` can be in a pending or ongoing `state`, and it can contain a `message` in case of failure.
+Other fields are `sessionId`, which maps to Cruise Control's `User-Task-ID`, and `targetReplicas`, which reflects the target replicas value (this may be different from .spec.replicas when the state is ongoing).
+
+- **Pending**: Not in Cruise Control's task queue (not yet sent or request error).
+  Cruise Control's task states: None.
+  ```sh
+  status:
+    conditions:
+    - lastTransitionTime: "2024-01-18T16:13:50.490918232Z"
+      status: "True"
+      type: Ready
+    observedGeneration: 2
+    replicasChange:
+      state: pending
+      targetReplicas: 2
+    topicName: my-topic
+  ```
+
+- **Ongoing**: In Cruise Control's task queue, but execution not started, or not completed. 
+  Cruise Control's task states: Active, InExecution.
+  ```sh
+  status:
+    conditions:
+    - lastTransitionTime: "2024-01-18T16:13:53.490918232Z"
+      status: "True"
+      type: Ready
+    observedGeneration: 3
+    replicasChange:
+      sessionId: 1aa418ca-53ed-4b93-b0a4-58413c4fc0cb
+      state: ongoing
+      targetReplicas: 2
+    topicName: my-topic
+  ```
+
+- **Completed**: Cruise Control's task execution completed (target replication factor reconciled).
+  Cruise Control's task states: Completed.
+  ```sh
+  status:
+    conditions:
+    - lastTransitionTime: "2024-01-18T16:13:58.490918232Z"
+      status: "True"
+      type: Ready
+    observedGeneration: 4
+    topicName: my-topic
+  ```
+
+- **Failed**: Cruise Control's request failed, the change remains pending, and it is periodically retried.
+  Cruise Control's task states: CompletedWithError.
+  ```sh
+  status:
+    conditions:
+    - lastTransitionTime: "2024-01-18T16:13:58.490918232Z"
+      status: "True"
+      type: Ready
+    observedGeneration: 4
+    replicasChange:
+      message: Change request failed, Cluster model not ready
+      state: pending
+      targetReplicas: 2
+    topicName: my-topic
+  ```
+
```

Comment:
I don't know if you want to include a diagram showing the allowed transitions. IIUC it would look like this:

```
+-->  Pending
|      |
|      V
|     Ongoing <------> Failed
|      |                |
|      V                |
+---- Completed <-------+
```